packages/grid/_modules_/grid/hooks/utils/useGridApiEventHandler.ts

```diff
@@ -44,15 +43,69 @@ export function useGridApiEventHandler<E extends GridEventsStr>(
       };
 
       subscription.current = apiRef.current.subscribeEvent(eventName, enhancedHandler, options);
-    }
 
-    return () => {
-      subscription.current?.();
+      cleanupTokensCounter += 1;
+      cleanupTokenRef.current = { cleanupToken: cleanupTokensCounter };
```

Comment:
Compared to https://github.com/Bnaya/use-dispose-uncommitted/blob/main/src/finalization-registry-based-impl.ts, why don't you have the `committed` key ?

Reply:
The hook that I based on has the option to pass an `onUncommittedCallback` callback. The `committed` key seems to be used to update the reference to this callback when the component rerenders before running the effects. It's a bit similar to https://github.com/mui-org/material-ui-x/blob/0b23cdd82d671fa7be52f9622de8cfe0ef1cd747/packages/grid/_modules_/grid/hooks/utils/useGridApiEventHandler.ts#L59